lib/scsi_defs.h

```diff
@@ -259,5 +261,32 @@ namespace scsi_defs {
   } ABSL_ATTRIBUTE_PACKED;
   static_assert(sizeof(ModeSense10Command) == 10);
 
+  // SCSI Reference Manual Table 148 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  enum class SelectReport : uint8_t {
+    kRestrictedMethods = 0x0,
+    kWellKnown = 0x1,
+    kAllLogical = 0x2
+  };
+
+  // SCSI Reference Manual Table 147 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  struct ReportLunsCommand {
+    OpCode op_code = OpCode::kReportLuns;
+    uint8_t reserved_1 : 8;
+    SelectReport select_report : 8;
+    uint32_t reserved_2 : 24;
+    uint32_t alloc_length : 32;
+    uint8_t reserved_3 : 8;
+    ControlByte control_byte;
+  } ABSL_ATTRIBUTE_PACKED;
+  static_assert(sizeof(ReportLunsCommand) == 12);
+
+  // SCSI Reference Manual Table 149 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  // This struct is a header for variable sized data
+  struct ReportLunsParamData {
+    uint32_t list_byte_length : 32;
```

Comment:
Bitfield annotations aren't necessary if you're using native sizes (8, 16, 32, 64 bits).

Reply:
Basim had found this stack overflow post that seems to indicate that the bitfields do cause some obscure difference where the size of the structure can be incorrectly packed if they're not used. Emily was able to replicate the discrepancy in our code base using the example in the post. Thoughts on this?
https://stackoverflow.com/questions/25822679/packed-bit-fields-in-c-structures-gcc

Reply:
Nothing wrong with consistency.  Comment withdrawn.